### PR TITLE
[FullStory] Use v2 specs

### PR DESCRIPTION
**Pull request by afharo**
When upgrading the FS snippet in Kibana (https://github.com/elastic/kibana/pull/189960), we noticed that FS has introduced a v2 spec in their APIs: https://developer.fullstory.com/browser/getting-started/#whats-new-in-version-2

1. Type annotations are dropped (instead of `version_str`, we can now report `version` and, optionally, annotate it via the `schema` attribute: https://developer.fullstory.com/browser/custom-properties/
2. APIs have been renamed (although v1 are still supported):
	- `FS.identify` -> `FS('setIdentify')`: https://developer.fullstory.com/browser/identification/identify-users/
	- `FS.setUserVars` -> `FS('setProperties', { type: 'user', properties });`: https://developer.fullstory.com/browser/identification/set-user-properties/
	- `FS.setVars` -> `FS('setProperties', { type: 'page', properties });`: https://developer.fullstory.com/browser/set-page-properties/
	- `FS.consent` -> `FS('setIdentity', { consent: true | false })`: https://developer.fullstory.com/browser/auto-capture/user-consent/
	- `FS.shutdown()` -> `FS('shutdown')`: https://developer.fullstory.com/browser/auto-capture/capture-data/
	- `FS.restart()` -> `FS('restart')`: https://developer.fullstory.com/browser/auto-capture/capture-data/
	- `FS.event` -> `FS('trackEvent', { name, properties })`: https://developer.fullstory.com/browser/capture-events/analytics-events/

V1 is still supported, but deprecated. Let's migrate before it's too late.

### CODE REVIEW

File: src/shippers/fullstory/src/types.ts [hunk 1]
```diff
@@ -6,61 +6,13 @@
  * Side Public License, v 1.
  */
 
+import type { FSApi } from '@fullstory/snippet';
```
Comment (afharo) [inline src/shippers/fullstory/src/types.ts:9]: Hopefully, from now on (+ the [renovate rules](https://github.com/elastic/ebt/pull/3)) we'll know when new versions are released.

**Review by afharo — COMMENTED** [review]
Apologies for the many files (but working without a linter was getting my nerves 😅).

The actual changes in this PR are in `src/shippers/fullstory/*`.

**Review by jloleysens — APPROVED** [review]
Thanks for getting ahead of this work @afharo ! Read through the code and looks great. I'm not sure how to e2e test these changes to ensure the shipper works as expected, but it looks like our end of the contract is being maintained.

(also what a beautiful new FS API 🤢 )

**Comment by afharo** [comment]
I'll wait for @Bamieh's review as well because I'm not sure how to publish this NPM module (permissions and tokens required)

**Review by Bamieh — APPROVED** [review]
LGTM

**Pull request merged**